src/lib.rs

```diff
@@ -377,33 +406,19 @@ impl Dependencies {
         flags.add(BuildFlag::RerunIfEnvChanged(EnvVariable::new_link(None)));
 
         for (name, _lib) in self.libs.iter() {
-            for var in EnvVariable::iter() {
-                let var = match var {
-                    EnvVariable::Lib(_) => EnvVariable::new_lib(name),
-                    EnvVariable::LibFramework(_) => EnvVariable::new_lib_framework(name),
-                    EnvVariable::SearchNative(_) => EnvVariable::new_search_native(name),
-                    EnvVariable::SearchFramework(_) => EnvVariable::new_search_framework(name),
-                    EnvVariable::Include(_) => EnvVariable::new_include(name),
-                    EnvVariable::NoPkgConfig(_) => EnvVariable::new_no_pkg_config(name),
-                    EnvVariable::BuildInternal(_) => EnvVariable::new_build_internal(Some(name)),
-                    EnvVariable::Link(_) => EnvVariable::new_link(Some(name)),
-                };
-                flags.add(BuildFlag::RerunIfEnvChanged(var));
-            }
+            EnvVariable::add_all_variants(&mut flags, name);
         }
 
         Ok(flags)
     }
 }
 
-#[derive(Error, Debug)]
+#[derive(Debug)]
 /// Error used in return value of `Config::add_build_internal` closures
 pub enum BuildInternalClosureError {
     /// `pkg-config` error
-    #[error(transparent)]
-    PkgConfig(#[from] pkg_config::Error),
```

Comment:
Also here the `From` impl is missing

Reply:
Same: what's the point of adding it if it works without it?

Reply:
It's more idiomatic, you're introducing an API change right now, and it does not work (you had to change `e.into()` into something else for it to work in all callers)

Reply:
FWIW, this is one of the reasons why `thiserror` is great. It makes it easier to create idiomatic error types that behave like they should. Such things can be easily forgotten.

Reply:
Ok...